src/host/PtySignalInputThread.cpp

```diff
@@ -237,44 +234,7 @@ void PtySignalInputThread::_DoSetWindowParent(const SetParentData& data)
     LockConsole();
     auto Unlock = wil::scope_exit([&] { UnlockConsole(); });
 
-    // If the client app hasn't yet connected, stash the new owner.
-    // We'll later (PtySignalInputThread::ConnectConsole) use the value
-    // to set up the owner of the conpty window.
-    if (!_consoleConnected)
-    {
-        _earlyReparent = data;
-        return;
-    }
-
-    const auto owner{ reinterpret_cast<HWND>(data.handle) };
-    // This will initialize s_interactivityFactory for us. It will also
-    // conveniently return 0 when we're on OneCore.
-    //
-    // If the window hasn't been created yet, by some other call to
-    // LocatePseudoWindow, then this will also initialize the owner of the
-    // window.
-    if (const auto pseudoHwnd{ ServiceLocator::LocatePseudoWindow(owner) })
-    {
-        // SetWindowLongPtrW may call back into the message handler and wait for it to finish,
-        // similar to SendMessageW(). If the conhost message handler is already processing and
-        // waiting to acquire the console lock, which we're currently holding, we'd deadlock.
-        // --> Release the lock now.
-        Unlock.reset();
-
-        // DO NOT USE SetParent HERE!
-        //
-        // Calling SetParent on a window that is WS_VISIBLE will cause the OS to
-        // hide the window, make it a _child_ window, then call SW_SHOW on the
-        // window to re-show it. SW_SHOW, however, will cause the OS to also set
-        // that window as the _foreground_ window, which would result in the
-        // pty's hwnd stealing the foreground away from the owning terminal
-        // window. That's bad.
-        //
-        // SetWindowLongPtr seems to do the job of changing who the window owner
-        // is, without all the other side effects of reparenting the window.
-        // See #13066
-        ::SetWindowLongPtrW(pseudoHwnd, GWLP_HWNDPARENT, reinterpret_cast<LONG_PTR>(owner));
-    }
+    ServiceLocator::SetPseudoWindowOwner(reinterpret_cast<HWND>(data.handle));

```

Comment:
The Azure PowerShell extension crashes the PowerShell process hard (not even an exception or anything) if you give it a `HWND_MESSAGE`. But my entire concept was based around that! That's why I moved all of the parent/owner code to `InteractivityFactory` too (below in this PR): A `HWND_MESSAGE` window cannot have a custom parent like this and so I had to store the owner in a member. I felt like `InteractivityFactory` should store it.

I left these changes in, because it IMO still makes the code easier to maintain: Now all of the parent/owner code is centralized in a single file.